lib/neo4j/active_node/query/query_proxy_eager_loading.rb

```diff
@@ -42,7 +45,24 @@ def paths(*prefix)
           end
 
           def process_hash(spec)
-            spec.each { |k, v| (self[k] ||= self.class.new(model, k)).add_spec(v) }
+            spec.each do |k, v|
+              rel_length = v.is_a?(Hash) ? v.delete(:rel_length) : nil
+              (self[k] ||= self.class.new(model, k, rel_length)).add_spec(v)
+            end
+          end
+
+          def process_string(spec)
+            paths = spec.split(',').collect { |path| path.split('.') }
+            paths.collect do |path|
+              path.reverse.inject({}) do |hash, rel|
+                if rel.include?('*')
+                  specs = rel.split('*')
+                  {specs.first.to_sym => hash.merge(rel_length: {min: 1, max: specs[1]})}
+                else
+                  {rel.to_sym => hash}
+                end
```

Comment:
the indirection via hash is unnecessary. Calling `new` directly from here is trivial.